server/controllers/survey.ts

```diff
@@ -0,0 +1,79 @@
+import { Request, Response } from 'express';
+
+import errors from '../lib/errors';
+
+export async function sendInAppSurveyResponse(
+  req: Request<
+    any,
+    any,
+    { surveyKey: string; responseId: string; score: string; text?: string; okToContact?: boolean }
+  >,
+  res: Response,
+  next,
+) {
+  const {
+    remoteUser,
+    body: { surveyKey, responseId, score, text, okToContact },
+  } = req;
+  if (!remoteUser) {
+    return next(new errors.Unauthorized());
+  }
+  if (!responseId) {
+    return next(new errors.BadRequest('Missing survey response id'));
+  }
+  const CODA_TOKEN = process.env.CODA_IN_APP_SURVEY_TOKEN;
+
+  if (!CODA_TOKEN) {
+    return next(new errors.ServerError('Missing CODA_IN_APP_SURVEY_TOKEN'));
+  }
+  const CODA_DOC_ID = 'nHLKv7oLV0';
```

Comment:
Some rate limits could prevent abuse here. Something as simple as:

```suggestion
  const rateLimit = new RateLimit(`survey-response-${remoteUser.id}`, 60, ONE_HOUR_IN_SECONDS);
  if (!(await rateLimit.registerCall())) {
    throw new RateLimitExceeded();
  }

  const CODA_DOC_ID = 'nHLKv7oLV0';
```

Reply:
Good call, added a rate limit